### PR TITLE
fix: update logging to log request_id and inventory_id

### PR DESCRIPTION
**Pull request by chenlizhong**
Update logging to log account,request_id and inventory_id

- Use logstash_formatter to generate json format log records for default logger
- Customize LogRecordFactory to record  account, request_id and inventory_id info in kafka

### CODE REVIEW

**Review by csams — CHANGES_REQUESTED** [review]
Consider something like contextvars and filters: https://docs.python.org/3/howto/logging-cookbook.html#use-of-contextvars since it's an established pattern and future proofs a little against race conditions.

**Comment by chenlizhong** [comment]
@csams 

Now the insights-engine is running in below kind of image, so the version of python is 3.6.8. But the contextVar module was introduced in 3.7.1    So can not use contextVar.  I will update the code to use Filter +  shared dict without contextVar. Do you agree ?

Of course, I prefer to upgrade the python to 3.9 (yum install python39).  I will raise this topic and email to related teams for discussion. But the discussion takes time. After python upgrade, we can update the code to use contextVar.

Docker file: 
> FROM registry.access.redhat.com/ubi8/ubi:latest
RUN yum -y install git autoconf automake gcc libtool file diffutils make python3 python3-pip && yum clean all

**Comment by chenlizhong** [comment]
@csams  The base image is updated to python 3.11 by MR https://github.com/RedHatInsights/insights-core-base/pull/10
So use filter and contextvar now. Please review again. Thanks.

File: insights_messaging/appbuilder.py [hunk 1]
```diff
@@ -133,8 +135,17 @@ def build_app(self):
         if log_config:
             logging.config.dictConfig(log_config)
         else:
+            handler = logging.StreamHandler(stream=sys.stdout)
+            handler.setFormatter(LogstashFormatterV1()) # to keep the same format with cloud env
             logging.basicConfig(level=logging.DEBUG)
-
+            logging.getLogger("insights.core.dr").setLevel(logging.ERROR)
```
Comment (csams) [inline insights_messaging/appbuilder.py:141]: Add a comment to explain what the handler at index 0 is and why removing it is necessary.

File: insights_messaging/consumers/kafka.py [hunk 2]
```diff
@@ -2,10 +2,18 @@
...
 from confluent_kafka import Consumer as ConfluentConsumer
-from insights_messaging.consumers import Consumer, Requeue
+from insights_messaging.consumers import Consumer, Requeue, archive_context_var
 
 log = logging.getLogger(__name__)
 
+def update_archive_context_ids(payload):
+    if payload and "platform_metadata" in payload and 'host' in payload:
+        payload_id_dict = {}
+        payload_id_dict['dummy_key'] = 'dummy_value'
```
Comment (csams) [inline insights_messaging/consumers/kafka.py:12]: What are `dummy_key` and `dummy_value` used for?

File: insights_messaging/appbuilder.py [hunk 3]
```diff
@@ -133,8 +135,17 @@ def build_app(self):
         if log_config:
             logging.config.dictConfig(log_config)
         else:
+            handler = logging.StreamHandler(stream=sys.stdout)
+            handler.setFormatter(LogstashFormatterV1()) # to keep the same format with cloud env
```
Comment (csams) [inline insights_messaging/appbuilder.py:139]: Why do we need `LogstashFormatter`? Is it to be consistent with other apps?

File: insights_messaging/consumers/kafka.py [hunk 4]
```diff
@@ -2,10 +2,18 @@
...
+        payload_id_dict['request_id'] = payload["platform_metadata"].get("request_id")
+        payload_id_dict['account'] = payload["platform_metadata"].get("account")
```
Comment (csams) [inline insights_messaging/consumers/kafka.py:14]: Is it okay to log the account? Do other applications do it?

**Review by csams — COMMENTED** [review]
It looks fine to me overall. I just have a few questions.

File: insights_messaging/consumers/kafka.py (hunk 4 above)
Comment (chenlizhong) [reply insights_messaging/consumers/kafka.py:14]: OK. We can remove it.

File: insights_messaging/appbuilder.py (hunk 3 above)
Comment (chenlizhong) [reply insights_messaging/appbuilder.py:139]: As the product, use the logstashformatter and write the log to aws log service.  So I update the log format for local testing (without aws)

**Review by csams — APPROVED** [review]
lgtm

**Pull request merged**